projects/ngx-charts-on-fhir/src/lib/fhir-chart/range-selector/range-selector.component.spec.ts

```diff
@@ -80,14 +81,10 @@ describe('RangeSelectorComponent', () => {
   });
 
   it('should display the range selector when one or more data layer is selected', () => {
-    component.layers = [{
-      id: '-109669932',
-      name: 'Blood Pressure',
-      category: 'vital-signs',
-      annotations: [],
-      selected: true,
-      enabled: true,
-    }];
+    mockLayerManager.selectedLayers$.subscribe((layers) => {
+      component.getMaxDateFromLayers(layers);
+      component.layers = layers as DataLayer[];
+    })
```

Comment:
why is this test subscribing to a mock observable?

Reply:
We are checking when one or more layer is selected then and only then do we need to display the range selector so in my mockLayerManager I have selected layer I am getting that and storing in  component.layer
Yes we do not need  `component.getMaxDateFromLayers(layers)`  here

Reply:
Yes, but the real component code is already subscribing to `selectedLayers$` and setting its own `layers` property. I don't see why the test code needs to duplicate this.

Generally, tests do not make direct calls to their own mocks. Mocks are provided to the real code in place of their real dependencies. The real code should call the mock (if it doesn't, then you don't need a mock).